CONTRIBUTION-GUIDE.md

```diff
@@ -0,0 +1,151 @@
+
+- **`docs/`** – Contains comprehensive documentation, including protocol specifications, developer guides, API references, and integration tutorials
+
+- **`explorer_backend/`** – Manages the backend services for the blockchain explorer, including data indexing, API endpoints, and real-time data processing
+
+- **`smart-contracts/`** – Hosts the `@nilfoundation/smart-contracts` package, comprising essential smart contracts for blockchain interactions
+
+(_Note: Some directories like **`nix`** are omitted for brevity, but contributors are encouraged to explore the full repository if necessary._)
+
+## How to Contribute
+
+This section describes in some detail how changes can be made and proposed with pull requests.
+
+### 1. Check Existing Issues
+
+Before starting work on a feature or bug fix, check existing [issues](https://github.com/NilFoundation/nil/issues). If none exist, consider opening an issue first to discuss your approach. This ensures alignment with project maintainers and avoids duplicate work.
+
+### 2. Fork the Repository
+
+- Click the **Fork** button on the repository's page
+- Clone your forked repository to your local machine:
+  ```sh
+  git clone https://github.com/your-username/nil.git
+  ```
+- Navigate to the project directory:
+  ```sh
+  cd nil
+  ```
+
+### 3. Create a Branch
+
+- Create a new branch for your feature or bug fix:
+  ```sh
+  git checkout -b feature-branch-name
+  ```
+
+### 4. Make Changes
+
+- Implement your changes following the project’s coding style
+- Ensure that all functional changes include appropriate tests
+- Test your changes thoroughly before submitting
+- Commit and sign your changes with a descriptive message:
+  ```sh
+  git add .
+  git commit -S -m "Add feature or fix description"
+  ```
+
+### 5. Push and Create a Pull Request
+
+- Push your changes to your forked repository:
+  ```sh
+  git push origin feature-branch-name
+  ```
+- Open a pull request from your forked repository to the main repository
+- Provide a clear description of your changes and the motivation behind them
+
+## Contribution Guidelines
+
+### PR Review and Merging Process
+
+- **PRs require approval and will be merged by a team member after signing, not directly by the author**. Maintainers will review and rebase them before merging
+- PRs must be signed by maintainers before merging
+- **The original authorship will be preserved**, but GPG signatures will be stripped, and the committer field will be changed
+- Direct merges from `master` are not allowed. Instead, PRs should be rebased if conflicts arise to maintain a clean and linear commit history. This approach helps avoid unnecessary merge commits, makes debugging easier, improves change tracking for reviews, reduces potential merge conflicts, and keeps the repository history structured and easy to navigate
+
+### Commit Message Guidelines
```

Comment:
Can we add PR title section where we explain that it is advisable to
1. add prefix of project before writing title (ex [explorer] drop unused file)
2. add reviewers
3. select project on the side
4. assign yourself to PR